### PR TITLE
[WIP] Update Annotation/Signature configuration and code refactoring for using zoom feature.

### PR DESCRIPTION
**Pull request by kegor**
Related PR with additional details and needed changes in UI https://github.com/groupdocs-total/GroupDocs.Total-Angular/pull/185.

### CODE REVIEW

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665286&bid=18545498)

File: src/Products/Annotation/Config/AnnotationConfiguration.cs [hunk 1]
```diff
@@ -11,26 +12,65 @@ namespace GroupDocs.Total.MVC.Products.Annotation.Config
...
-        private bool isZoom = true;
-        private bool isFitWidth = true;
+        [JsonProperty]
+        private string FilesDirectory = "DocumentSamples/Annotation";
+
+        [JsonProperty]
+        private string defaultDocument = "";
+
+        [JsonProperty]
+        private int preloadPageCount = 0;
```
Comment (egorovpavel) [inline src/Products/Annotation/Config/AnnotationConfiguration.cs:22]: Codacy Issue found: [Remove this initialization to 'preloadPageCount', the compiler will do that for you.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665286)

Comment (bobkovalex) [inline src/Products/Annotation/Config/AnnotationConfiguration.cs:22]: Codacy Issue found: [Remove this initialization to 'preloadPageCount', the compiler will do that for you.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665285)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665285&bid=18545497)

File: src/Products/Annotation/Controllers/AnnotationApiController.cs [hunk 2]
```diff
@@ -88,30 +88,30 @@ public HttpResponseMessage LoadFileTree(AnnotationPostedDataEntity fileTreeReque
...
+                    FileInfo fileInfo = new FileInfo(fd.Guid);
                     // check if current file/folder is temp directory or is hidden
-                    FileInfo fileInfo = new FileInfo(fileDescription.guid);
-                    if (tempDirectoryName.ToLower().Equals(fileDescription.guid) || fileInfo.Attributes.HasFlag(FileAttributes.Hidden))
-                    {
-                        // ignore current file and skip to next one
-                        continue;
-                    }
-                    else
+                    if (!(tempDirectoryName.ToLower().Equals(Path.GetFileName(fd.Guid).ToLower()) ||
```
Comment (bobkovalex) [inline src/Products/Annotation/Controllers/AnnotationApiController.cs:93]: Codacy Issue found: [Define the locale to be used in this string operation.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665285)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665286&bid=18545498)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 8
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665285&bid=18545497)

File: src/Products/Annotation/Controllers/AnnotationApiController.cs [hunk 3]
```diff
@@ -88,30 +87,30 @@ public HttpResponseMessage LoadFileTree(AnnotationPostedDataEntity fileTreeReque
...
                 {
-                    FileDescriptionEntity fileDescription = new FileDescriptionEntity
-                    {
-                        guid = fd.Guid
-                    };
+                    FileInfo fileInfo = new FileInfo(fd.Guid);
                     // check if current file/folder is temp directory or is hidden
-                    FileInfo fileInfo = new FileInfo(fileDescription.guid);
-                    if (tempDirectoryName.ToLower().Equals(fileDescription.guid) || fileInfo.Attributes.HasFlag(FileAttributes.Hidden))
+                    if (!(tempDirectoryName.ToLower().Equals(Path.GetFileName(fd.Guid).ToLower()) ||
```
Comment (bobkovalex) [inline src/Products/Annotation/Controllers/AnnotationApiController.cs:92]: Codacy Issue found: [Define the locale to be used in this string operation.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665285)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 8
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665285&bid=18545497)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665286&bid=18545498)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 11
- Added 3
           

Clones added
============
- src/Products/Signature/Controllers/SignatureApiController.cs  1
         

Clones removed
==============
+ src/Products/Signature/Config/SignatureConfiguration.cs  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665285&bid=18545497)

File: src/Products/Signature/Loader/SignatureLoader.cs [hunk 4]
```diff
@@ -15,16 +16,18 @@ public class SignatureLoader
     {
         public string CurrentPath;
         public Common.Config.GlobalConfiguration globalConfiguration;
+        private DirectoryUtils DirectoryUtils;
```
Comment (bobkovalex) [inline src/Products/Signature/Loader/SignatureLoader.cs:19]: Codacy Issue found: [Make 'DirectoryUtils' 'readonly'.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665285)

File: src/Products/Signature/Config/SignatureConfiguration.cs [hunk 5]
```diff
@@ -11,21 +12,51 @@ namespace GroupDocs.Total.MVC.Products.Signature.Config
...
+
+        [JsonProperty]
         public string DefaultDocument = "";
+
+        [JsonProperty]
         public string DataDirectory = "";
-        public int PreloadPageCount = 0;
+
+        [JsonProperty]
+        public int PreloadPageCount;
```
Comment (bobkovalex) [inline src/Products/Signature/Config/SignatureConfiguration.cs:25]: Codacy Issue found: [Make this field 'private' and encapsulate it in a 'public' property.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665285)

Comment (bobkovalex) [inline src/Products/Signature/Config/SignatureConfiguration.cs:25]: Codacy Issue found: [Make 'PreloadPageCount' private.](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665285)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 10
- Added 3
           

Clones added
============
- src/Products/Signature/Controllers/SignatureApiController.cs  1
         

Clones removed
==============
+ src/Products/Signature/Config/SignatureConfiguration.cs  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665286&bid=18545498)

File: src/Products/Signature/Loader/SignatureLoader.cs (hunk 4 above)
Comment (egorovpavel) [inline src/Products/Signature/Loader/SignatureLoader.cs:19]: Codacy Issue found: [Make 'DirectoryUtils' 'readonly'.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665286)

File: src/Products/Signature/Config/SignatureConfiguration.cs (hunk 5 above)
Comment (egorovpavel) [inline src/Products/Signature/Config/SignatureConfiguration.cs:25]: Codacy Issue found: [Make 'PreloadPageCount' private.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665286)

Comment (egorovpavel) [inline src/Products/Signature/Config/SignatureConfiguration.cs:25]: Codacy Issue found: [Make this field 'private' and encapsulate it in a 'public' property.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665286)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 35
           

Clones added
============
- src/Products/Signature/Controllers/SignatureApiController.cs  1
         

Clones removed
==============
+ src/Products/Signature/Loader/SignatureLoader.cs  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665285&bid=18545497)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 34
           

Clones added
============
- src/Products/Signature/Controllers/SignatureApiController.cs  1
         

Clones removed
==============
+ src/Products/Signature/Loader/SignatureLoader.cs  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665286&bid=18545498)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 35
           

Clones added
============
- src/Products/Signature/Controllers/SignatureApiController.cs  1
         

Clones removed
==============
+ src/Products/Signature/Loader/SignatureLoader.cs  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665285&bid=18545497)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 34
           

Clones added
============
- src/Products/Signature/Controllers/SignatureApiController.cs  1
         

Clones removed
==============
+ src/Products/Signature/Loader/SignatureLoader.cs  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665286&bid=18545498)

**Comment by egorovpavel** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 34
           

Clones added
============
- src/Products/Signature/Controllers/SignatureApiController.cs  1
         

Clones removed
==============
+ src/Products/Signature/Loader/SignatureLoader.cs  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/GroupDocs/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665286&bid=18545498)

**Comment by bobkovalex** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 35
           

Clones added
============
- src/Products/Signature/Controllers/SignatureApiController.cs  1
         

Clones removed
==============
+ src/Products/Signature/Loader/SignatureLoader.cs  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/groupdocs-total/GroupDocs.Total-for-NET-MVC/pullRequest?prid=5665285&bid=18545497)

**Pull request merged**